src/index.ts

```diff
@@ -79,16 +96,25 @@ export class PromiseQueue {
 						reject(e);
 						return;
 					}
+					this.metrics.emit('queueTime', durationSince(enqueueTime));
+					this.metrics.emit('dequeue');
+					const serviceStartTime = process.hrtime();
 					const result = await fn();
+					this.metrics.emit('serviceTime', durationSince(serviceStartTime));
```

Comment:
These never get emitted in the case of a failure, so it's probably better to put them in a finally

Reply:
Ah, good catch

Reply:
Now fixed